<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Converters — convert • mia</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Converters — convert"><meta property="og:description" content="For convenience, a few functions are available to convert BIOM, DADA2 and
phyloseq objects to
TreeSummarizedExperiment
objects, and
TreeSummarizedExperiment
objects to phyloseq objects."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mia</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.13.32</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/mia.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/microbiome/mia/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Converters</h1>
    <small class="dont-index">Source: <a href="https://github.com/microbiome/mia/blob/devel/R/convertFromBIOM.R" class="external-link"><code>R/convertFromBIOM.R</code></a>, <a href="https://github.com/microbiome/mia/blob/devel/R/convertFromDADA2.R" class="external-link"><code>R/convertFromDADA2.R</code></a>, <a href="https://github.com/microbiome/mia/blob/devel/R/convertFromPhyloseq.R" class="external-link"><code>R/convertFromPhyloseq.R</code></a>, and 1 more</small>
    <div class="hidden name"><code>convert.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>For convenience, a few functions are available to convert BIOM, DADA2 and
phyloseq objects to
<code><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/TreeSummarizedExperiment-class.html" class="external-link">TreeSummarizedExperiment</a></code>
objects, and
<code><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/TreeSummarizedExperiment-class.html" class="external-link">TreeSummarizedExperiment</a></code>
objects to phyloseq objects.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">convertFromBIOM</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  prefix.rm <span class="op">=</span> <span class="va">removeTaxaPrefixes</span>,</span>
<span>  removeTaxaPrefixes <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  rank.from.prefix <span class="op">=</span> <span class="va">rankFromPrefix</span>,</span>
<span>  rankFromPrefix <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  artifact.rm <span class="op">=</span> <span class="va">remove.artifacts</span>,</span>
<span>  remove.artifacts <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">convertToBIOM</span><span class="op">(</span><span class="va">x</span>, assay.type <span class="op">=</span> <span class="st">"counts"</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'SummarizedExperiment'</span></span>
<span><span class="fu">convertToBIOM</span><span class="op">(</span><span class="va">x</span>, assay.type <span class="op">=</span> <span class="st">"counts"</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">convertFromDADA2</span><span class="op">(</span><span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">convertFromPhyloseq</span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">convertToPhyloseq</span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'SummarizedExperiment'</span></span>
<span><span class="fu">convertToPhyloseq</span><span class="op">(</span><span class="va">x</span>, assay.type <span class="op">=</span> <span class="st">"counts"</span>, assay_name <span class="op">=</span> <span class="cn">NULL</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'TreeSummarizedExperiment'</span></span>
<span><span class="fu">convertToPhyloseq</span><span class="op">(</span><span class="va">x</span>, tree.name <span class="op">=</span> <span class="va">tree_name</span>, tree_name <span class="op">=</span> <span class="st">"phylo"</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-x">x<a class="anchor" aria-label="anchor" href="#arg-x"></a></dt>
<dd><p>a
<code><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/TreeSummarizedExperiment-class.html" class="external-link">TreeSummarizedExperiment</a></code>
object</p></dd>


<dt id="arg-prefix-rm">prefix.rm<a class="anchor" aria-label="anchor" href="#arg-prefix-rm"></a></dt>
<dd><p><code>TRUE</code> or <code>FALSE</code>: Should
taxonomic prefixes be removed? The prefixes is removed only from detected
taxa columns meaning that <code>rank.from.prefix</code> should be enabled in the most cases.
(default <code>prefix.rm = FALSE</code>)</p></dd>


<dt id="arg-removetaxaprefixes">removeTaxaPrefixes<a class="anchor" aria-label="anchor" href="#arg-removetaxaprefixes"></a></dt>
<dd><p>Deprecated. Use <code>prefix.rm</code> instead.</p></dd>


<dt id="arg-rank-from-prefix">rank.from.prefix<a class="anchor" aria-label="anchor" href="#arg-rank-from-prefix"></a></dt>
<dd><p><code>TRUE</code> or <code>FALSE</code>: If file does not have
taxonomic ranks on feature table, should they be scraped from prefixes?
(default <code>rank.from.prefix = FALSE</code>)</p></dd>


<dt id="arg-rankfromprefix">rankFromPrefix<a class="anchor" aria-label="anchor" href="#arg-rankfromprefix"></a></dt>
<dd><p>Deprecated.Use <code>rank.from.prefix</code> instead.</p></dd>


<dt id="arg-artifact-rm">artifact.rm<a class="anchor" aria-label="anchor" href="#arg-artifact-rm"></a></dt>
<dd><p><code>TRUE</code> or <code>FALSE</code>: If file have
some taxonomic character naming artifacts, should they be removed.
(default <code>artifact.rm = FALSE</code>)</p></dd>


<dt id="arg-remove-artifacts">remove.artifacts<a class="anchor" aria-label="anchor" href="#arg-remove-artifacts"></a></dt>
<dd><p>Deprecated. Use <code>artifact.rm</code> instead.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Additional arguments. For <code>convertFromDADA2</code>, see
<code>mergePairs</code> function for more details.</p></dd>


<dt id="arg-assay-type">assay.type<a class="anchor" aria-label="anchor" href="#arg-assay-type"></a></dt>
<dd><p>A single character value for selecting the
<code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></code> to be
included in the phyloseq object that is created.
(By default: <code>assay.type = "counts"</code>)</p></dd>


<dt id="arg-assay-name">assay_name<a class="anchor" aria-label="anchor" href="#arg-assay-name"></a></dt>
<dd><p>a single <code>character</code> value for specifying which
assay to use for calculation.
(Please use <code>assay.type</code> instead. At some point <code>assay_name</code>
will be disabled.)</p></dd>


<dt id="arg-tree-name">tree.name<a class="anchor" aria-label="anchor" href="#arg-tree-name"></a></dt>
<dd><p>a single <code>character</code> value for specifying which
tree will be included in the phyloseq object that is created,
(By default: <code>tree.name = "phylo"</code>)</p></dd>


<dt id="arg-tree-name">tree_name<a class="anchor" aria-label="anchor" href="#arg-tree-name"></a></dt>
<dd><p>Deprecated. Use <code>tree.name</code> instead.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p><code>convertFromBIOM</code> returns an object of class
<code><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/TreeSummarizedExperiment-class.html" class="external-link">TreeSummarizedExperiment</a></code></p>
<p><code>convertFromDADA2</code> returns an object of class
<code><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/TreeSummarizedExperiment-class.html" class="external-link">TreeSummarizedExperiment</a></code></p>
<p><code>convertFromPhyloseq</code> returns an object of class
<code><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/TreeSummarizedExperiment-class.html" class="external-link">TreeSummarizedExperiment</a></code></p>
<p><code>convertToPhyloseq</code> returns an object of class
<code><a href="https://rdrr.io/pkg/phyloseq/man/phyloseq-class.html" class="external-link">phyloseq</a></code></p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p><code>convertFromBIOM</code> coerces a BIOM object to a
<code><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/TreeSummarizedExperiment-class.html" class="external-link">TreeSummarizedExperiment</a></code>
object.</p>
<p><code>convertFromBIOM</code> coerces a
<code><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/TreeSummarizedExperiment-class.html" class="external-link">TreeSummarizedExperiment</a></code>
object to a <code><a href="https://rdrr.io/pkg/biomformat/man/biom-class.html" class="external-link">biom</a></code> object.</p>
<p><code>convertFromDADA2</code> is a wrapper for the
<code>mergePairs</code> function from the <code>dada2</code> package.
A count matrix is constructed via
<code>makeSequenceTable(mergePairs(...))</code> and rownames are dynamically
created as <code>ASV(N)</code> with <code>N</code> from 1 to <code>nrow</code> of the count
tables. The colnames and rownames from the output of <code>makeSequenceTable</code>
are stored as <code>colnames</code> and in the <code>referenceSeq</code> slot of the
<code>TreeSummarizedExperiment</code>, respectively.</p>
<p><code>convertFromPhyloseq</code> converts <code>phyloseq</code>
objects into
<code><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/TreeSummarizedExperiment-class.html" class="external-link">TreeSummarizedExperiment</a></code> objects.
All data stored in a <code>phyloseq</code> object is transferred.</p>
<p><code>convertToPhyloseq</code> creates a phyloseq object from a
<code><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/TreeSummarizedExperiment-class.html" class="external-link">TreeSummarizedExperiment</a></code>
object. By using <code>assay.type</code>, it is possible to specify which table
from <code>assay</code> is added to the phyloseq object.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="importQIIME2.html">importQIIME2</a></code>
<code><a href="importMothur.html">importMothur</a></code></p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Leo Lahti and Tuomas Borman. Contact: <a href="microbiome.github.io">microbiome.github.io</a></p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Convert BIOM results to a TreeSE</span></span></span>
<span class="r-in"><span><span class="co"># Load biom file</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/joey711/biomformat/" class="external-link">biomformat</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">biom_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"rich_dense_otu_table.biom"</span>,</span></span>
<span class="r-in"><span>                         package <span class="op">=</span> <span class="st">"biomformat"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Make TreeSE from BIOM object</span></span></span>
<span class="r-in"><span><span class="va">biom_object</span> <span class="op">&lt;-</span> <span class="fu">biomformat</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/biomformat/man/read_biom.html" class="external-link">read_biom</a></span><span class="op">(</span><span class="va">biom_file</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu">convertFromBIOM</span><span class="op">(</span><span class="va">biom_object</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Convert TreeSE object to BIOM</span></span></span>
<span class="r-in"><span><span class="va">biom</span> <span class="op">&lt;-</span> <span class="fu">convertToBIOM</span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">### Coerce DADA2 results to a TreeSE object</span></span></span>
<span class="r-in"><span><span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"dada2"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span>  <span class="va">fnF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"sam1F.fastq.gz"</span>, package<span class="op">=</span><span class="st">"dada2"</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="va">fnR</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"sam1R.fastq.gz"</span>, package<span class="op">=</span><span class="st">"dada2"</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="va">dadaF</span> <span class="op">&lt;-</span> <span class="fu">dada2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/dada2/man/dada.html" class="external-link">dada</a></span><span class="op">(</span><span class="va">fnF</span>, selfConsist<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="va">dadaR</span> <span class="op">&lt;-</span> <span class="fu">dada2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/dada2/man/dada.html" class="external-link">dada</a></span><span class="op">(</span><span class="va">fnR</span>, selfConsist<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>  <span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu">convertFromDADA2</span><span class="op">(</span><span class="va">dadaF</span>, <span class="va">fnF</span>, <span class="va">dadaR</span>, <span class="va">fnR</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="va">tse</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required namespace: dada2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Initializing error rates to maximum possible estimate.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> selfConsist step 1 .</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Convergence after  4  rounds.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Initializing error rates to maximum possible estimate.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> selfConsist step 1 .</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    selfConsist step 4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Convergence after  4  rounds.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> class: TreeSummarizedExperiment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> dim: 4 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> metadata(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> assays(1): counts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rownames(4): ASV01 ASV02 ASV03 ASV04</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowData names(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colnames: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colData names(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> reducedDimNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> mainExpName: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> altExpNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowLinks: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowTree: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colLinks: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colTree: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> referenceSeq: a DNAStringSet (4 sequences)</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">### Coerce a phyloseq object to a TreeSE object</span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"phyloseq"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>, package<span class="op">=</span><span class="st">"phyloseq"</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="fu">convertFromPhyloseq</span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">enterotype</span>, package<span class="op">=</span><span class="st">"phyloseq"</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="fu">convertFromPhyloseq</span><span class="op">(</span><span class="va">enterotype</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">esophagus</span>, package<span class="op">=</span><span class="st">"phyloseq"</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="fu">convertFromPhyloseq</span><span class="op">(</span><span class="va">esophagus</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required namespace: phyloseq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> class: TreeSummarizedExperiment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> dim: 58 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> metadata(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> assays(1): counts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rownames(58): 59_8_22 59_5_13 ... 65_9_9 59_2_6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowData names(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colnames(3): B C D</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colData names(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> reducedDimNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> mainExpName: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> altExpNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowLinks: a LinkDataFrame (58 rows)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowTree: 1 phylo tree(s) (58 leaves)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colLinks: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colTree: NULL</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">### Coerce a TreeSE object to a phyloseq object</span></span></span>
<span class="r-in"><span><span class="co"># Get tse object</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="va">GlobalPatterns</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Create a phyloseq object from it</span></span></span>
<span class="r-in"><span><span class="va">phy</span> <span class="op">&lt;-</span> <span class="fu">convertToPhyloseq</span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">phy</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> phyloseq-class experiment-level object</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_table()   OTU Table:         [ 19216 taxa and 26 samples ]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sample_data() Sample Data:       [ 26 samples by 7 sample variables ]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tax_table()   Taxonomy Table:    [ 19216 taxa by 7 taxonomic ranks ]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> phy_tree()    Phylogenetic Tree: [ 19216 tips and 19215 internal nodes ]</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># By default the chosen table is counts, but if there are other tables,</span></span></span>
<span class="r-in"><span><span class="co"># they can be chosen with assay.type.</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Counts relative abundances table</span></span></span>
<span class="r-in"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="transformAssay.html">transformAssay</a></span><span class="op">(</span><span class="va">tse</span>, method <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">phy2</span> <span class="op">&lt;-</span> <span class="fu">convertToPhyloseq</span><span class="op">(</span><span class="va">tse</span>, assay.type <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">phy2</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> phyloseq-class experiment-level object</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> otu_table()   OTU Table:         [ 19216 taxa and 26 samples ]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sample_data() Sample Data:       [ 26 samples by 7 sample variables ]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tax_table()   Taxonomy Table:    [ 19216 taxa by 7 taxonomic ranks ]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> phy_tree()    Phylogenetic Tree: [ 19216 tips and 19215 internal nodes ]</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Felix G.M. Ernst, Sudarshan A. Shetty, Tuomas Borman, Leo Lahti.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

      </footer></div>






  </body></html>

